.github/workflows/nfr.yml

```diff
@@ -35,104 +38,142 @@ permissions:
-        echo "GKE_CLUSTER_ZONE=us-east1-b" >> vars.env
-        echo "GKE_CLUSTER_REGION=us-east1" >> vars.env
-        echo "GKE_PROJECT=${{ secrets.GCP_PROJECT_ID }}" >> vars.env
-        echo "GKE_SVC_ACCOUNT=${{ secrets.GCP_SERVICE_ACCOUNT }}" >> vars.env
-        echo "GKE_NODES_SERVICE_ACCOUNT=${{ secrets.GKE_NODES_SERVICE_ACCOUNT }}" >> vars.env
-        echo "IMAGE=projects/debian-cloud/global/images/debian-11-bullseye-v20240213" >> vars.env
-        echo "NETWORK_TAGS=nfr-tests-${{ github.run_id }}" >> vars.env
-        echo "NGF_REPO=nginxinc" >> vars.env
-        echo "NGF_BRANCH=${{ github.ref_name }}" >> vars.env
-        echo "SOURCE_IP_RANGE=$(curl -sS -4 icanhazip.com)/32" >> vars.env
-        echo "ADD_VM_IP_AUTH_NETWORKS=true" >> vars.env
-        echo "PLUS_ENABLED=${{ inputs.nginx_plus }}" >> vars.env
-        echo "GINKGO_LABEL=" >> vars.env
-        echo "NGF_VERSION=${{ inputs.version }}" >> vars.env
-        echo "GKE_NUM_NODES=12" >> vars.env
-        echo "GKE_MACHINE_TYPE=n2d-standard-16" >> vars.env
-
-    - name: Create GKE cluster
-      working-directory: ./tests
-      run:
-        make create-gke-cluster CI=true
-
-    - name: Create and setup VM
-      working-directory: ./tests
-      run:
-        make create-and-setup-vm
-
-    - name: Run Tests
-      working-directory: ./tests
-      run: |
-        if ${{ inputs.test_label != 'all' }}; then
-        sed -i '/^GINKGO_LABEL=/s/=.*/="${{ inputs.test_label }}"/' "scripts/vars.env" && make nfr-test;
-        else
-        make nfr-test;
-        fi
-
-    - name: Cleanup
-      working-directory: ./tests
-      if: always()
-      run: |
-        bash scripts/cleanup-vm.sh true
-        make delete-gke-cluster
-        rm -rf scripts/vars.env
-
-    - name: Open a PR with the results
-      uses: peter-evans/create-pull-request@c5a7806660adbe173f04e3e038b0ccdcd758773c # v6.1.0
-      with:
-        token: ${{ secrets.NGINX_PAT }}
-        commit-message: NFR Test Results for NGF version ${{ inputs.version }} ${{ inputs.nginx_plus == true && '(Plus)' || ''}}
-        author: nginx-bot <integrations@nginx.com>
-        committer: nginx-bot <integrations@nginx.com>
-        branch: tests/nfr-tests-${{ inputs.version }}${{ inputs.nginx_plus == true && '-plus' || ''}}
-        delete-branch: true
-        title: NFR Test Results for NGF version ${{ inputs.version }} ${{ inputs.nginx_plus == true && '(Plus)' || ''}}
-        add-paths: |
-          tests/results/
-        body: |
-          Update with NFR test results for NGF version ${{ inputs.version }} ${{ inputs.nginx_plus == true && '(Plus)' || ''}}
-          - Auto-generated by the NFR tests workflow run ${{ github.run_id }}
-          - Tests ran using Docker image tag ${{ inputs.image_tag }}
-          - ${{ inputs.test_label }} test(s) ran
-          - NGINX Plus enabled: ${{ inputs.nginx_plus }}
-        labels: |
-          tests
-        assignees: ${{ github.actor }}
-        draft: true
```

Comment:
If we add analysis to each PR, we should probably keep this as a draft initially.

Reply:
My thinking was that once this PR is open, it's ready for us to review. Also we won't get notified for a draft.
Why do you think it's better to keep it as a draft?

Reply:
I think the logic was just that we don't accidentally approve and merge before we've analyzed. Having it as a draft makes it clear that changes need to be made.

Reply:
Though for scheduled runs, I don't know if we need the overhead of adding summaries every time. Probably more important for the release ones.

Reply:
oooh I forgot about the summaries. Maybe just put it in draft when it's not a schedule?

Reply:
Yeah, that's what I'm thinking.